pkg/csi/cinder/openstack/openstack_volumes.go

```diff
@@ -180,6 +184,10 @@ func (os *OpenStack) WaitDiskAttached(instanceID string, volumeID string) error
 func (os *OpenStack) DetachVolume(instanceID, volumeID string) error {
 	volume, err := os.GetVolume(volumeID)
 	if err != nil {
+		if cpoerrors.IsNotFound(err) {
```

Comment:
I think we need to handle this in ControllerUnpublishVolume and return NOT_FOUND error instead as per csi spec ,  https://github.com/container-storage-interface/spec/blob/master/spec.md#controllerunpublishvolume-errors

Reply:
+1

Reply:
I added some comment in original bug. if we follow the spec, the part of the bug is  invalid
as the bug is requesting to ignore the error ...